.github/workflows/cloud_chatops.yaml

```diff
@@ -25,17 +28,80 @@ jobs:
 
       - name: Install dependencies
         run: |
+          cd cloud_chatops
           python -m pip install --upgrade pip
-          cd cloud_chatops && pip install -r requirements.txt
+          pip install -r requirements.txt
 
       - name: Analyse with pylint
-        run: pylint cloud_chatops --rcfile=cloud_chatops/.pylintrc
+        run: |
+          cd cloud_chatops
+          pylint src --rcfile=.pylintrc
+          pylint tests --rcfile=.pylintrc
+
+      - name: Run tests
+        run: |
+          cd cloud_chatops
+          python3 -m pytest tests
 
       - name: Run tests and collect coverage
-        run: cd cloud_chatops && python3 -m pytest tests --cov-report xml:coverage.xml --cov
+        run: |
+          cd cloud_chatops
+          python3 -m pytest tests --cov-report xml:coverage.xml --cov
 
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v4
         with:
           token: ${{secrets.CODECOV_TOKEN}}
-          files: ./cloud_chatops/coverage.xml
+          files: cloud_chatops/coverage.xml
+
+  push_dev_image_harbor:
+    runs-on: ubuntu-latest
+    needs: test_and_lint
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Harbor
+        uses: docker/login-action@v3
+        with:
+          registry: harbor.stfc.ac.uk
+          username: ${{ secrets.STAGING_HARBOR_USERNAME }}
+          password: ${{ secrets.STAGING_HARBOR_TOKEN }}
+
+      - name: Build and push to staging project
+        uses: docker/build-push-action@v6
+        with:
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          push: true
+          context: "{{defaultContext}}:cloud_chatops"
+          tags: "harbor.stfc.ac.uk/stfc-cloud-staging/cloud-chatops"
```

Comment:
This needs some sort of tag with a version, else it will push to `latest`
We use the SHA elsewhere